src/System.Threading.Channels/src/System/Collections/Generic/Dequeue.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Collections.Generic
+{
+    /// <summary>Provides a double-ended queue data structure.</summary>
+    /// <typeparam name="T">Type of the data stored in the dequeue.</typeparam>
+    [DebuggerDisplay("Count = {_size}")]
+    internal sealed class Dequeue<T>
```

Comment:
Should we make it public? and expose from collections package?

Reply:
We could. That should be a separate proposal/design/etc. Right now this is just the minimal support I needed.